### DISCUSSION TITLE
Error handling when requesting auth token

### DISCUSSION THREAD

**Discussion by alexaxb** [AdhocAdam/smletsexchangeconnector#402] [Ideas]
It would be nice to add a try/catch around the oauth token request down below, so if for some reason you are not allowed to request the token, it gets logged as an error.

$response = Invoke-RestMethod -Uri $azureTokenURL -Method "POST" -Body $ReqTokenBody

#instead of a username/password, use the OAuth access_token as the means to authenticate to Exchange
$exchangeService.Url = [System.Uri]$ExchangeEndpoint
$exchangeService.Credentials = [Microsoft.Exchange.WebServices.Data.OAuthCredentials]($response.Access_Token)

**Comment by AdhocAdam** [comment]
> It would be nice to add a try/catch around the oauth token request down below, so if for some reason you are not allowed to request the token, it gets logged as an error.

_and_ triggers a Custom Event? I can think of a couple opportunities here. 😄 

In any case, if you want to make a Pull Request against the dev branch I'm always open to reviewing. Especially as I'm working towards [v5](https://github.com/AdhocAdam/smletsexchangeconnector/pull/401).
